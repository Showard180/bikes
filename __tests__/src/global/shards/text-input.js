import 'react-native';
import React from 'react';
import TextInput from '../../../../src/global/shards/text-input';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders', () => {
  const tree = renderer.create(
    <TextInput />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders when given title and placeholder props', () => {
  const props = {
    text: 'Placehoder',
    title: 'Text input test'
  }

  const tree = renderer.create(
    <TextInput {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders when given onChangeText props', () => {
  const props = {
    interactiveFunction: () => true
  }

  const tree = renderer.create(
    <TextInput {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('behaves correctly when an input is given', () => {
  let string = undefined;
  const props = {
    interactiveFunction: ({text}) => string = text
  }

  const tree = renderer.create(
    <TextInput {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
  tree.children[1].children[0].props.onChangeText('Testing');
  expect(string).toEqual('Testing');
})
