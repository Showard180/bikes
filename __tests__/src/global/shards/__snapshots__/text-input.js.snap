// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`behaves correctly when an input is given 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 15,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 40,
        "marginVertical": 10,
      }
    }
  >
    <TextInput
      keyboardType={undefined}
      onChangeText={[Function]}
      placeholder={undefined}
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`renders 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 15,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 40,
        "marginVertical": 10,
      }
    }
  >
    <TextInput
      keyboardType={undefined}
      onChangeText={[Function]}
      placeholder={undefined}
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`renders when given onChangeText props 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 15,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 40,
        "marginVertical": 10,
      }
    }
  >
    <TextInput
      keyboardType={undefined}
      onChangeText={[Function]}
      placeholder={undefined}
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`renders when given title and placeholder props 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 15,
      }
    }
  >
    Text input test
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 40,
        "marginVertical": 10,
      }
    }
  >
    <TextInput
      keyboardType={undefined}
      onChangeText={[Function]}
      placeholder="Placehoder"
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
